src/main/java/Duke/Task/TaskList.java

```diff
@@ -0,0 +1,187 @@
+     */
+    public void findTasks(String keyword) {
+        int count = 0;
+
+        for (int i = 0; i < tasks.size(); i++) {
+            String message = tasks.get(i).getTaskInfo();
+            if (message.contains(keyword)) {
+                count++;
+                System.out.println(count + "." + message);
+            }
+        }
+
+        if (count == 0) {
+            System.out.println("Sorry. There is no tasks matching the keyword you give me!! :(");
+        }
+    }
+
+    /**
+     * Marks a specific task as done.
+     *
+     * @param index Integer indicates the index for the task
+     */
+    public void markDone(int index) {
+        this.tasks.get(index).markDone();
+    }
+
+    /**
+     * Deletes a specific task.
+     *
+     * @param index Integer indicates the index for the task
+     */
+    public void delete(int index) {
+        this.tasks.remove(index);
+    }
+
+    /**
+     * Adds a task to the TaskLists. This method will automatically decide which type of the
+     * task is added to the list.
+     *
+     * @param taskType String message indicates the task type.
+     * @param task String message indicates the task info.
+     * @param time String message indicates the time info.
+     * @throws DukeException Throws when a task cannot be created or added to the TaskList.
+     */
+    public void add(String taskType, String task, String time) throws DukeException {
+        Parser parser = new Parser("");
+
+        LocalDateTime parsedTime = parser.parseTime(time);
+        OperationType[] taskTypes = OperationType.values();
+        for (OperationType t : taskTypes) {
+            if (t.toString().equals(taskType)) {
+                Task newTask = t.assignTaskType(t, task, parsedTime);
+                tasks.add(newTask);
+                break;
+            }
+        }
+
+    }
+
+    /**
+     * Returns a specific task users refer to.
+     *
+     * @param index An integer indicates the index of the task.
+     * @return Task users refer to
+     * */
+    public Task get(int index) {
+        return this.tasks.get(index);
+    }
+
+
+    /**
+     * Returns the size of the TaskList
+     *
+     * @return Integer indicates the size of the TaskList.
+     */
+    public int size() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Detects if the index taking in is invalid or not.
+     *
+     * @param index Integer indicates the index of the task.
+     * @throws DukeException Throws when the index is invalid.
+     */
+    public void detectIndex(int index) throws DukeException {
+        if (index < 0 || index >= this.tasks.size()) {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but the index is invalid :-(");
+        }
+    }
+
+    /**
+     * Enum of all types of operations that is able to execute.
+     * It also contains a method AssignTask Type to find the specific type of task to create.
+     */
+    public enum OperationType {
```

Comment:
https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/oop.html#enumerations
Nice try using enum, might be good to CAPS everything cos its all final values